### PR TITLE
chore(deps): bump to rust-bitcoin 0.32.5

### PR DESCRIPTION
**Pull request by storopoli**
Related to https://github.com/rust-bitcoin/rust-bitcoin/pull/3729.

Keeping this as a draft since I am planning to do PRs in more `BitVM` org crates to bump `rust-bitcoin` to 0.32.5 (latest).

I had to change one `Message` deprecation from `rust-secp256k1` when updating to `0.29.1` which is the version that `rust-bitcoin` `0.32.5` re-exports.

Also update `musig2` to play nice with `rust-bitcoin@0.32.5`.

Blocked by:
- https://github.com/BitVM/rust-bitcoin-script-stack/pull/4
- https://github.com/BitVM/rust-esplora-client/pull/1

### CODE REVIEW

**Comment by lucidLuckylee** [comment]
@storopoli I merged the blocking PRs.

**Comment by storopoli** [comment]
This is ready

File: src/bridge/transactions/pre_signed_musig2.rs [hunk 1]
```diff
@@ -67,7 +70,8 @@ pub fn push_nonce<T: PreSignedTransaction + PreSignedMusig2Transaction>(
 }
 
 pub fn get_nonce_message(nonce: &PubNonce) -> Message {
-    Message::from_hashed_data::<bitcoin::hashes::sha256::Hash>(nonce.to_bytes().as_slice())
+    let nonce_hash = sha256::Hash::hash(nonce.to_bytes().as_slice());
+    Message::from_digest_slice(nonce_hash.as_ref()).expect("Failed to create nonce message")
```
Comment (storopoli) [inline src/bridge/transactions/pre_signed_musig2.rs:74]: I decide to `.expect()` here to not break the API by changing the function signature to return a `Result`.

**Comment by lucidLuckylee** [comment]
Thank you!

**Pull request merged**